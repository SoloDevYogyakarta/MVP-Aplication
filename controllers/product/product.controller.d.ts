/// <reference types="multer" />
import { Request, Response } from 'express';
import { DyanmicQuery } from '../../validators/query/product.query';
import { ProductRepository } from '../../repository/product/product.repository';
import { ProductService } from '../../services/product/product.service';
import { CustomRequest } from '../../types/custom-request.type';
export declare class ProductController {
    private readonly service;
    private readonly repository;
    constructor(service: ProductService, repository: ProductRepository);
    create(req: CustomRequest, res: Response, files: Express.Multer.File[]): Promise<Response<any, Record<string, any>>>;
    update(req: CustomRequest, res: Response, files: Express.Multer.File[]): Promise<Response<any, Record<string, any>>>;
    destroy(req: CustomRequest, res: Response): Promise<Response<any, Record<string, any>>>;
    list(query: DyanmicQuery, res: Response): Promise<Response<any, Record<string, any>>>;
    detail(req: Request, res: Response): Promise<Response<any, Record<string, any>>>;
}
