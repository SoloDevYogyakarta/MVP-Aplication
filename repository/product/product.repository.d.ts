import { ProductBasicEntity, ProductBasicInstance } from '../../database/entities/products/basic-entity/basic-entity';
export declare class ProductRepository {
    findOne(public_id: string): Promise<ProductBasicInstance>;
    findAll(query: ProductBasicEntity): Promise<ProductBasicInstance[]>;
}
