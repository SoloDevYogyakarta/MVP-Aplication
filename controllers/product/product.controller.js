"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductController = void 0;
const common_1 = require("@nestjs/common");
const product_query_1 = require("../../validators/query/product.query");
const guards_middleware_1 = require("../../middleware/guards.middleware");
const product_repository_1 = require("../../repository/product/product.repository");
const product_service_1 = require("../../services/product/product.service");
const platform_express_1 = require("@nestjs/platform-express");
const multer_1 = require("../../utils/multer/multer");
let ProductController = class ProductController {
    constructor(service, repository) {
        this.service = service;
        this.repository = repository;
    }
    async create(req, res, files) {
        const result = await this.service.create(req.body, req.user.data.public_id, files);
        return res.status(result.status).json(result);
    }
    async update(req, res, files) {
        const result = await this.service.update(req.body, req.params.id, files);
        return res.status(result.status).json(result);
    }
    async destroy(req, res) {
        const result = await this.service.destroy(req.params.id);
        return res.status(result.status).json(result);
    }
    async list(query, res) {
        const result = await this.repository.findAll(query);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
    async detail(req, res) {
        const result = await this.repository.findOne(req.params.id);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
};
exports.ProductController = ProductController;
__decorate([
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    (0, common_1.Post)(),
    (0, common_1.UseInterceptors)((0, platform_express_1.FilesInterceptor)('files', 5, { storage: multer_1.diskStorage })),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __param(2, (0, common_1.UploadedFiles)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object, Array]),
    __metadata("design:returntype", Promise)
], ProductController.prototype, "create", null);
__decorate([
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    (0, common_1.Post)(':id'),
    (0, common_1.UseInterceptors)((0, platform_express_1.FilesInterceptor)('files', 5, { storage: multer_1.diskStorage })),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __param(2, (0, common_1.UploadedFiles)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object, Array]),
    __metadata("design:returntype", Promise)
], ProductController.prototype, "update", null);
__decorate([
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    (0, common_1.Delete)(':id'),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], ProductController.prototype, "destroy", null);
__decorate([
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    (0, common_1.Get)(),
    __param(0, (0, common_1.Query)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [product_query_1.DyanmicQuery, Object]),
    __metadata("design:returntype", Promise)
], ProductController.prototype, "list", null);
__decorate([
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    (0, common_1.Get)(':id'),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], ProductController.prototype, "detail", null);
exports.ProductController = ProductController = __decorate([
    (0, common_1.Controller)('product'),
    __metadata("design:paramtypes", [product_service_1.ProductService,
        product_repository_1.ProductRepository])
], ProductController);
//# sourceMappingURL=product.controller.js.map