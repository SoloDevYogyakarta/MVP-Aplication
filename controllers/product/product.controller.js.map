{"version":3,"file":"product.controller.js","sourceRoot":"","sources":["../../../src/controllers/product/product.controller.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;AAAA,2CAYwB;AAExB,wEAAoE;AACpE,0EAA+D;AAC/D,oFAAgF;AAChF,4EAAwE;AAExE,+DAA4D;AAC5D,sDAAwD;AAIjD,IAAM,iBAAiB,GAAvB,MAAM,iBAAiB;IAC5B,YACmB,OAAuB,EACvB,UAA6B;QAD7B,YAAO,GAAP,OAAO,CAAgB;QACvB,eAAU,GAAV,UAAU,CAAmB;IAC7C,CAAC;IAKE,AAAN,KAAK,CAAC,MAAM,CACH,GAAkB,EAClB,GAAa,EACH,KAA4B;QAE7C,OAAO,CAAC,GAAG,CAAC,GAAG,CAAC,IAAI,CAAC,CAAC;QACtB,MAAM,MAAM,GAAG,MAAM,IAAI,CAAC,OAAO,CAAC,MAAM,CACtC,GAAG,CAAC,IAAqC,EACzC,GAAG,CAAC,IAAI,CAAC,IAAI,CAAC,SAAS,EACvB,KAAK,CACN,CAAC;QACF,OAAO,GAAG,CAAC,MAAM,CAAC,MAAM,CAAC,MAAM,CAAC,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC;IAChD,CAAC;IAKK,AAAN,KAAK,CAAC,MAAM,CACH,GAAY,EACZ,GAAa,EACH,KAA4B;QAE7C,MAAM,MAAM,GAAG,MAAM,IAAI,CAAC,OAAO,CAAC,MAAM,CAAC,GAAG,CAAC,IAAI,EAAE,GAAG,CAAC,MAAM,CAAC,EAAE,EAAE,KAAK,CAAC,CAAC;QACzE,OAAO,GAAG,CAAC,MAAM,CAAC,MAAM,CAAC,MAAM,CAAC,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC;IAChD,CAAC;IAIK,AAAN,KAAK,CAAC,OAAO,CAAQ,GAAY,EAAS,GAAa;QACrD,MAAM,MAAM,GAAG,MAAM,IAAI,CAAC,OAAO,CAAC,OAAO,CAAC,GAAG,CAAC,MAAM,CAAC,EAAE,CAAC,CAAC;QACzD,OAAO,GAAG,CAAC,MAAM,CAAC,MAAM,CAAC,MAAM,CAAC,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC;IAChD,CAAC;IAIK,AAAN,KAAK,CAAC,IAAI,CAAU,KAAmB,EAAS,GAAa;QAC3D,MAAM,MAAM,GAAG,MAAM,IAAI,CAAC,UAAU,CAAC,OAAO,CAAC,KAAK,CAAC,CAAC;QACpD,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,EAAE,CAAC,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC;IAChD,CAAC;IAIK,AAAN,KAAK,CAAC,MAAM,CAAQ,GAAY,EAAS,GAAa;QACpD,MAAM,MAAM,GAAG,MAAM,IAAI,CAAC,UAAU,CAAC,OAAO,CAAC,GAAG,CAAC,MAAM,CAAC,EAAE,CAAC,CAAC;QAC5D,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,EAAE,CAAC,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC;IAChD,CAAC;CACF,CAAA;AAvDY,8CAAiB;AAStB;IAHL,IAAA,kBAAS,EAAC,6BAAS,CAAC;IACpB,IAAA,aAAI,GAAE;IACN,IAAA,wBAAe,EAAC,IAAA,mCAAgB,EAAC,OAAO,EAAE,CAAC,EAAE,EAAE,OAAO,EAAE,oBAAW,EAAE,CAAC,CAAC;IAErE,WAAA,IAAA,YAAG,GAAE,CAAA;IACL,WAAA,IAAA,YAAG,GAAE,CAAA;IACL,WAAA,IAAA,sBAAa,GAAE,CAAA;;;;+CASjB;AAKK;IAHL,IAAA,kBAAS,EAAC,6BAAS,CAAC;IACpB,IAAA,aAAI,EAAC,KAAK,CAAC;IACX,IAAA,wBAAe,EAAC,IAAA,mCAAgB,EAAC,OAAO,EAAE,CAAC,EAAE,EAAE,OAAO,EAAE,oBAAW,EAAE,CAAC,CAAC;IAErE,WAAA,IAAA,YAAG,GAAE,CAAA;IACL,WAAA,IAAA,YAAG,GAAE,CAAA;IACL,WAAA,IAAA,sBAAa,GAAE,CAAA;;;;+CAIjB;AAIK;IAFL,IAAA,kBAAS,EAAC,6BAAS,CAAC;IACpB,IAAA,eAAM,EAAC,KAAK,CAAC;IACC,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;gDAGxC;AAIK;IAFL,IAAA,kBAAS,EAAC,6BAAS,CAAC;IACpB,IAAA,YAAG,GAAE;IACM,WAAA,IAAA,cAAK,GAAE,CAAA;IAAuB,WAAA,IAAA,YAAG,GAAE,CAAA;;qCAApB,4BAAY;;6CAGtC;AAIK;IAFL,IAAA,kBAAS,EAAC,6BAAS,CAAC;IACpB,IAAA,YAAG,EAAC,KAAK,CAAC;IACG,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;+CAGvC;4BAtDU,iBAAiB;IAD7B,IAAA,mBAAU,EAAC,SAAS,CAAC;qCAGQ,gCAAc;QACX,sCAAiB;GAHrC,iBAAiB,CAuD7B"}