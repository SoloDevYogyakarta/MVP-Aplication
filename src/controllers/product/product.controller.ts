import {
  Controller,
  Delete,
  Get,
  HttpStatus,
  Post,
  Query,
  Req,
  Res,
  UploadedFiles,
  UseGuards,
  UseInterceptors,
} from '@nestjs/common';
import { Request, Response } from 'express';
import { DyanmicQuery } from '../../validators/query/product.query';
import { AuthGuard } from '../../middleware/guards.middleware';
import { ProductRepository } from '../../repository/product/product.repository';
import { ProductService } from '../../services/product/product.service';
import { CustomRequest } from '../../types/custom-request.type';
import { FilesInterceptor } from '@nestjs/platform-express';
import { diskStorage } from '../../utils/multer/multer';
import { createpath } from '../../utils/system/system';
import { omit } from 'lodash';

@Controller('product')
export class ProductController {
  constructor(
    private readonly service: ProductService,
    private readonly repository: ProductRepository,
  ) {}

  @UseGuards(AuthGuard)
  @Post()
  @UseInterceptors(FilesInterceptor('files', 5, { storage: diskStorage }))
  async create(
    @Req() req: CustomRequest,
    @Res() res: Response,
    @UploadedFiles() files: Express.Multer.File[],
  ) {
    const result = await this.service.create(
      req.body,
      req.user.data.public_id,
      files,
    );
    createpath('../../database/dataTxt/basic-http-entity.txt', result.result);
    return res.status(result.status).json(omit(result, ['result']));
  }

  @UseGuards(AuthGuard)
  @Post(':id')
  @UseInterceptors(FilesInterceptor('files', 5, { storage: diskStorage }))
  async update(
    @Req() req: CustomRequest,
    @Res() res: Response,
    @UploadedFiles() files: Express.Multer.File[],
  ) {
    const result = await this.service.update(
      req.body as unknown as { [key: string]: string },
      req.params.id,
      files,
      req.user.data.public_id,
    );
    return res.status(result.status).json(result);
  }

  @UseGuards(AuthGuard)
  @Delete(':id')
  async destroy(@Req() req: CustomRequest, @Res() res: Response) {
    const result = await this.service.destroy(
      req.params.id,
      req.user.data.public_id,
    );
    return res.status(result.status).json(result);
  }

  @UseGuards(AuthGuard)
  @Get()
  async list(@Query() query: DyanmicQuery, @Res() res: Response) {
    const result = await this.repository.findAll(query);
    return res.status(HttpStatus.OK).json(result);
  }

  @UseGuards(AuthGuard)
  @Get(':id')
  async detail(@Req() req: Request, @Res() res: Response) {
    const result = await this.repository.findOne(req.params.id);
    return res.status(HttpStatus.OK).json(result);
  }
}
