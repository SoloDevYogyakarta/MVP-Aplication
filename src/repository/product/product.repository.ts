import { Injectable } from '@nestjs/common';
import {
  productBasicAssociate,
  productBasicAttribute,
  productBasicInclude,
} from '../../database/associates/basic-associate/basic-associate';
import { ProductBasicInstance } from '../../database/entities/products/product-basic-entity/product-basic-entity';
import { Op } from 'sequelize';
import { DyanmicQuery } from 'src/validators/query/product.query';
import { some } from 'lodash';

@Injectable()
export class ProductRepository {
  async findOne(public_id: string): Promise<ProductBasicInstance> {
    const result = await productBasicAssociate.findOne({
      where: { public_id },
      attributes: productBasicAttribute,
      include: productBasicInclude,
    });
    return result;
  }

  async findAll(query: DyanmicQuery): Promise<ProductBasicInstance[]> {
    let where = {},
      datas = [];
    Object.keys(query).filter((key) => {
      datas = [...datas, { [key]: { [Op.iLike]: query[key] } }];
    });
    if (some(query)) {
      where = { [Op.or]: datas };
    }
    return await productBasicAssociate.findAll({
      where: where,
      attributes: productBasicAttribute,
      include: productBasicInclude,
    });
  }
}
