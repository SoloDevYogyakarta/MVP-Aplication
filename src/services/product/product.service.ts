import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import {
  ProductBasicEntity,
  productBasicEntity,
} from '../../database/entities/products/basic-entity/basic-entity';
import { pick } from 'lodash';
import { joinpath, removepath } from '../../utils/system/system';
import {
  ProductPriceEntity,
  productpriceEntity,
} from '../../database/entities/products/price-entity/price-entity';
import {
  productStockEntity,
  ProductStockEntity,
} from '../../database/entities/products/stock-entity/stock-entity';
import sizeOf from 'image-size';
import { fileEntity } from '../../database/entities/commons/file-entity/file-entity';
import { joinEntity } from '../../database/entities/commons/join-entity/join-entity';
import { nanoid } from 'nanoid';
import { userEntity } from '../../database/entities/authenticates/user-entity/user-entity';

type Product = ProductBasicEntity & {
  price?: ProductPriceEntity;
  stock?: ProductStockEntity;
};

@Injectable()
export class ProductService {
  async create(body: Product, user_id: string, files: Express.Multer.File[]) {
    await this.isAdmin(user_id);
    const create = await productBasicEntity.create(
      pick({ ...body, public_id: nanoid(), user_id }, [
        'public_id',
        'name',
        'status',
        'condition',
        'shortdesc',
        'main_stock',
        'reserve_stock',
        'user_id',
      ]),
    );
    create.save();
    const dataPrice =
      typeof body.price === 'string' ? JSON.parse(body.price) : body.price;
    const price = await productpriceEntity.create(
      pick({ ...dataPrice, product_id: create.public_id }, [
        'value',
        'currency',
        'product_id',
      ]),
    );
    const dataStock =
      typeof body.stock === 'string' ? JSON.parse(body.stock) : body.stock;
    const stock = await productStockEntity.create({
      ...dataStock,
      product_id: create.public_id,
    });
    stock.save();
    price.save();
    if (files?.length) {
      for (const file of files) {
        file.path = `/assets${file.path.split('assets')[1]}`;
        const f = await fileEntity.create({
          public_id: nanoid(),
          ...file,
          filepath: file.path,
          ...sizeOf(joinpath(`../../assets/${file.filename}`)),
        });
        f.save();
        const join = await joinEntity.create({
          public_id: nanoid(),
          source_id: create.public_id,
          foreign_id: f.public_id,
        });
        join.save();
      }
    }
    return {
      result: create,
      status: HttpStatus.CREATED,
      message: 'Product has been create',
    };
  }

  async update(
    body: Product,
    public_id: string,
    files: Express.Multer.File[],
    user_id: string,
  ) {
    await this.isAdmin(user_id);
    const findOne = await productBasicEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.NOT_FOUND,
          message: 'Not Found',
        },
        HttpStatus.NOT_FOUND,
      );
    }
    findOne.update(
      pick(body, [
        'name',
        'status',
        'condition',
        'shortdesc',
        'main_stock',
        'reserve_stock',
      ]),
      { where: { public_id } },
    );
    await productpriceEntity.update(
      typeof body.price === 'string' ? JSON.parse(body.price) : body.price,
      {
        where: { product_id: findOne.public_id },
      },
    );
    await productStockEntity.update(
      typeof body.stock === 'string' ? JSON.parse(body.stock) : body.stock,
      {
        where: { product_id: findOne.public_id },
      },
    );
    if (files?.length) {
      for (const file of files) {
        const join = await joinEntity.findOne({
          where: { source_id: findOne.public_id },
        });
        const f = await fileEntity.findOne({
          where: { public_id: join.foreign_id },
        });
        if (f.filepath) {
          try {
            removepath(`../..${f.filepath}`);
          } catch (err) {
            // empty
          }
        }
        f.destroy();
        join.destroy();
        file.path = `/assets${file.path.split('assets')[1]}`;
        const cf = await fileEntity.create({
          public_id: nanoid(),
          ...file,
          filepath: file.path,
          ...sizeOf(joinpath(`../../assets/${file.filename}`)),
        });
        cf.save();
        const cj = await joinEntity.create({
          source_id: findOne.public_id,
          foreign_id: cf.public_id,
        });
        cj.save();
      }
    }
    return { status: HttpStatus.OK, message: 'Product has been update' };
  }

  async destroy(public_id: string, user_id: string) {
    await this.isAdmin(user_id);
    const findOne = await productBasicEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.NOT_FOUND,
          message: 'Not Found',
        },
        HttpStatus.NOT_FOUND,
      );
    }
    await productpriceEntity.destroy({
      where: { product_id: findOne.public_id },
    });
    await productStockEntity.destroy({
      where: { product_id: findOne.public_id },
    });
    const joins = await joinEntity.findAll({
      where: { source_id: findOne.public_id },
    });
    if (joins.length) {
      for (const join of joins) {
        const file = await fileEntity.findOne({
          where: { public_id: join.foreign_id },
        });
        if (file.filepath) {
          try {
            removepath(`../..${file.filepath}`);
          } catch (err) {
            // empty
          }
        }
        file.destroy();
        join.destroy();
      }
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Product has been delete' };
  }

  async isAdmin(public_id: string): Promise<void> {
    const findOne = await userEntity.findOne({ where: { public_id } });
    if (findOne?.role !== 'admin') {
      throw new HttpException('false', HttpStatus.INTERNAL_SERVER_ERROR);
    }
  }
}
