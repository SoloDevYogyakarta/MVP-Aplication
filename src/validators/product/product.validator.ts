export class Variant {
  public_id: string;
  type: string;
  name: string;
  desc: string;
}

export class ProductField {
  type?: string;
  date?: Date;
  mechanis_name?: string;
  desc?: string;
  variants: string;
}
