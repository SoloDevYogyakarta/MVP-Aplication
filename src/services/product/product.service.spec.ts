import { ProductService } from './product.service';

describe('ProductService', () => {
  let service: ProductService;

  beforeEach(() => {
    service = new ProductService();
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('render correctly', () => expect(service).toMatchSnapshot());
});
