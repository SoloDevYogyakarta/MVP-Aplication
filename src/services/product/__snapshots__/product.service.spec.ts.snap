// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductService render correctly 1`] = `ProductService {}`;
